ML/AliExternalBDT.cxx

```diff
@@ -120,24 +122,33 @@ bool AliExternalBDT::LoadLightGBMModel(std::string path) {
 
 bool AliExternalBDT::LoadModelLibrary(std::string path) {
   const int status = TreelitePredictorLoad(path.data(), 1, &fPredictor);
+
+  TreelitePredictorQueryResultSizeSingleInst(fPredictor, &fOutSize);
+  TreelitePredictorQueryNumFeature(fPredictor, &fNumFeatures);
+
   if (status != 0) {
     std::cerr << "Library loading failed" << std::endl;
     return false;
   }
   return true;
 }
 
-double AliExternalBDT::Predict(double *features, int size, bool useRawScore) {
+bool AliExternalBDT::Predict(double *features, int size, std::vector<double> &outputScores, bool useRawScore) {
```

Comment:
I would not use a vector but an array since its dimension is known a priory when you write the task I think. However, I don't have a strong opinion here.

Instead the casting of the features from double to float and the results from float to double do not make much sense. Probably we should move to float since this is the intrinsic precision of the model application. We can address this in another PR eventually.

Reply:
For the first comment, it is a bit more complex to use an array because the size of the output is not necessarily known a priori, and there might be cases in which this is not even considered in a task (i.e. if you only call IsSelctedMulticlass without the output scores). One should define an array with a not defined size that depends on the size of the output score taken from treelite, so personally I would keep a vector at least for the time being.

Regarding the second comment, I completely agree, I did not change it not to modify all the tasks that use this method but we can consider to do it in the future.